src/main/webapp/app/exercises/programming/manage/update/programming-exercise-update.component.html

```diff
@@ -1,594 +1,687 @@
-<form name="editForm" role="form" novalidate #editForm="ngForm" (keydown.enter)="isEventInsideTextArea($event)">
-    <h4 *ngIf="!isImport && !programmingExercise.id" id="jhi-programming-exercise-heading-create" jhiTranslate="artemisApp.programmingExercise.home.generateLabel">
-        Generate new Programming Exercise
-    </h4>
-    <h4 *ngIf="!isImport && programmingExercise.id" id="jhi-programming-exercise-heading-edit" jhiTranslate="artemisApp.programmingExercise.home.editLabel">
-        Edit Programming Exercise
-    </h4>
-    <h4 *ngIf="isImport" id="jhi-programming-exercise-heading-import" jhiTranslate="artemisApp.programmingExercise.home.importLabel">Import Programming Exercise</h4>
-    <div class="update-programming-exercise">
-        <div class="form-group" [hidden]="isImport || !programmingExercise.id">
-            <label for="id" jhiTranslate="global.field.id">ID</label>
-            <input type="text" class="form-control" id="id" name="id" [(ngModel)]="programmingExercise.id" readonly />
-        </div>
-        <div class="form-group">
-            <div>
-                <label class="label-narrow" jhiTranslate="artemisApp.exercise.title" for="field_title">Title</label>
-                <jhi-help-icon placement="auto" text="artemisApp.programmingExercise.titleTooltip"></jhi-help-icon>
+<div>
+    <jhi-programming-exercise-update-wizard
```

Comment:
That many `@inputs` possibly stem from [prop drilling](https://kentcdodds.com/blog/prop-drilling). A possible solution would be to supply this data through a service

Reply:
I moved the inputs for the different steps to their own types (like you suggested for the programming languages). That reduces the inputs here by a lot